### DISCUSSION TITLE
How about check type of predicate with generic

### DISCUSSION THREAD

**Discussion by linlinyang** [vitest-dev/vitest#5440] [Q&A]
``` typescript
const includes = <T>(arr: T[], val: unknow): val is T => arr.includes(val);
```
Is there any way test `includes` function，`expectTypeOf(includes).guards` get `never` type, and `assertType(includes([1,2,3], 1))` just check return type but not predicate.

**Comment by mrazauskas** [comment]
Might be worth asking in the [`expect-type`](https://github.com/mmkal/expect-type) repo. Their APIs are re-exported here.
